dev-tools/jenkins_ci.sh

```diff
@@ -146,6 +146,10 @@ case `uname` in
             IVY_REPO="${GIT_TOPLEVEL}/../ivy"
             mkdir -p "${IVY_REPO}/maven/org/elasticsearch/ml/ml-cpp/$VERSION"
             cp "../build/distributions/ml-cpp-$VERSION-linux-$HARDWARE_ARCH.zip" "${IVY_REPO}/maven/org/elasticsearch/ml/ml-cpp/$VERSION/ml-cpp-$VERSION.zip"
+            # Since this is all local, for simplicity, cheat with the dependencies/no-dependencies split
```

Comment:
Not sure I follow this bit.

Reply:
This is for the step of testing ml-cpp PRs where we run a subset of Elasticsearch integration tests on Linux using the ml-cpp code that's changed by the PR.

We were already cheating here, because we renamed the zip file containing just the code for Linux on the current architecture to the name of the zip file that's supposed to contain the code for all platforms.

Since the test is taking place on the local machine and the Elasticsearch build will "download" the newly built code from an Ivy repo on the local disk, we might as well just put all the files in one zip and have an empty zip for the other one - the combination of the two zip files will still contain all the necessary files.

Partly this depends on how the Elasticsearch side works.  I've opened https://github.com/elastic/elasticsearch/pull/82268 with my first attempt at this and it doesn't work at all - if I point what's currently in that PR at a local Ivy repo containing two zip files then it doesn't currently incorporate either of them into the build.  Please let me know if you have any ideas for how to make it work.

Reply:
Yeah, the project substitution stuff is going to be tricky since you're saying "replace both these things with the same project artifact". There are a few different things we could do here, to solve this problem which we can discuss separate from this PR. I think for now the simplest thing is just to use the "full" artifact when integrating with the Elasticsearch build.